### ISSUE TITLE
[SECURITY ISSUE] Breaking change by adding jinja templates also introduces code execution

### ISSUE THREAD

**Issue by palako** [instructor-ai/instructor#1132]
We've been pined to an old version of instruction due to continuous breaking bugs being introduced. Yesterday we tried to run with the latest to see if things were any better and to our surprise things broke quite severely. It seems that version 1.5 switched to using jinja templates by default, meaning, all of our prompts are now treated as jinja templates.

Our prompts were already coming from jinja templates, plus, for our application, we often include file contents in prompts, some of which are jinja, and we use the cookiecutter library, that uses jinja variables as filenames, so even adding filenames to a prompt now produces a failure.

As per the above, things are broken beyond any chance of using it, so we need to go and pin back to something prior to 1.5. There doesn't seem to be an option to not use the new templating functionality.

Besides the broken behaviour, this now means that anything that makes it into a template is being evaluated and rendered as jinja, which allows python code as part of its syntax. Due to the lack of user input sanitation, this is now a severe security problem allowing for code execution for any one passing user input to instructor. This is even more severe considering that this change has been introduced transparently, so applications using instructor were not expected to be making things jinja safe and they haven't been told to start doing so either.

The request is for the jinja functionality to be either completely removed until implemented properly, or at the very least disable it and only enable it when explicitly requested, the API for template should be optional and completely separated from the way messages where parsed before.

**Comment by gokturkDev** [comment]
@jxnl  
if this is truly the case please add prominent disclaimers to the repo and docs please!

**Comment by ivanleomk** [comment]
@palako could you show an example of this? Just trying to understand what an exploit might look like.

**Comment by ivanleomk** [comment]
This is solved in #1168 so closing this issue for now

**Comment by a3lem** [comment]
For reasons I can't understand, instructor applying template rendering causes this error:

```
-> raise TypeError("Can't compile non template nodes")
```

Context: using genai with a two-part content message (text + image).

Since I don't need (or want) Instructor to handle template rendering for me, I patched the `apply_template()` function such that it becomes a non-op. That solved the crashes.

```python
def patch_instructor():
    import instructor.templating
    instructor.templating.apply_template = lambda text, context: text

patch_instructor()
```
